### DISCUSSION TITLE
concourse web start failed due to session-signing-key

### DISCUSSION THREAD

**Discussion by liya2017** [concourse/concourse#6888] [Help & Support]
Hi guys
Really need your help.
I am trying to set up concourse work node on my vm.But I met some errors. Let me show some main info for you.
1. vm is ubuntu-focal-20.04-amd64 on aws
2. concourse version is `6.7.6` 
3. concourse quickstart is ok with below command 
```
concourse quickstart \
--add-local-user test:123456 \
--main-team-local-user test \
--external-url http://roens-ci.test.com \
--worker-work-dir /opt/concourse/worker \
--postgres-user concourse \
--postgres-password 123456
```

4. generated the keys with below command:
```
sudo ssh-keygen -t rsa -N '' -f /etc/concourse/tsa_host_key 
sudo ssh-keygen -t rsa -N '' -f /etc/concourse/worker_key 
sudo ssh-keygen -t rsa -N '' -f /etc/concourse/session_signing_key
cp /etc/concourse/worker_key.pub /etc/concourse/authorized_worker_keys
```

5. want to start concourse worker node with below command, but met errors:
`concourse web --add-local-user liya:123456 --main-team-local-user liya --session-signing-key /etc/concourse/session_signing_key  --tsa-host-key /etc/concourse/tsa_host_key --tsa-authorized-keys /etc/concourse --external-url http://concourse.ckb.com --postgres-user concourse --postgres-password 123456`

```
error: invalid argument for flag `--session-signing-key' (expected *flag.PrivateKey): asn1: structure error: tags don't match (16 vs {class:1 tag:15 length:112 isCompound:true}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} pkcs8 @w32blaster 
```

Could you please give me some hands?

**Comment by taylorsilva** [comment]
You can generate the keys with the concourse binary as well.
```
concourse generate-key -t rsa -f /etc/concourse/session_signing_key
```

Maybe the same is happening to you but my `ssh-keygen` keeps making key-pairs in what appears to be an older format, based on what I just searched up: https://tls.mbed.org/kb/cryptography/asn1-key-structures-in-der-and-pem

ssh-keygen: generating keys in PKCS#1 format
concourse binary: makes and expects keys in PKCS#8 format (it seems?)

End of the error you posted also mentions PKCS#8: `pkcs8`

**Reply by taylorsilva** [reply]
Yup, ssh-keygen generates keys in PKCS#1 format.
https://stackoverflow.com/questions/21718193/making-git-more-secure-how-to-gen-ssh-keys-in-pkcs8-format-with-the-openssl-ex

**Reply by liya2017** [reply]
It's work now for me by your point. Thanks for your help !
